### PR TITLE
5 - Block_Users Handler

### PR DESCRIPTION
**Pull request by TriAnh1995**
**Request:  As a user, I need an API to block updates from an email address**

- Initializing Block_Users Handler

### CODE REVIEW

File: api/internal/handler/validate.go [hunk 1]
```diff
@@ -36,6 +36,13 @@ func (e FriendsList) validate() error {
 	}
 	return nil
 }
+func (b Block) validate() error {
+	emails := []string{b.Requester, b.Target}
+	if err := validateEmails(emails); err != nil {
+		return err
+	}
+	return nil
```
Comment (nhan1603) [inline api/internal/handler/validate.go]: Can we specify the error for requester and target or they both have the same error message ?

Comment (TriAnh1995) [reply api/internal/handler/validate.go]: Yes we could specify the error message for requester and target.
I though it was unnecessary but i see your point. I'll look into it!

File: api/internal/handler/validate.go [hunk 2]
```diff
@@ -36,6 +36,20 @@ func (e FriendsList) validate() error {
...
 	return nil
 }
+func (b Block) validate() error {
+	emails := []string{b.Requester, b.Target}
+	if (len(emails) != 2) || (emails[0] == emails[1]) {
+		return errors.New("Please insert two different emails")
+	}
+
+	if requesterError := validateEmail(b.Requester); requesterError != nil {
+		return errors.New("Something wrong with the Requester email: " + requesterError.Error())
```
Comment (nhan1603) [inline api/internal/handler/validate.go]: Can change the message error to : `"Invalid format of requester email: "`

File: api/internal/handler/validate.go [hunk 3]
```diff
@@ -36,6 +36,20 @@ func (e FriendsList) validate() error {
...
+	}
+	if targetError := validateEmail(b.Target); targetError != nil {
+		return errors.New("Something wrong with the Target email: " + targetError.Error())
```
Comment (nhan1603) [inline api/internal/handler/validate.go]: Same here

**Pull request merged**